### PR TITLE
Add github webhook signing secret to constructor; add checkSignature() to parseHook()

### PR DESCRIPTION
**Pull request by tkyi**
#As per St.John's suggestions to decouple Screwdriver API from Github
- adds Github webhook signing secret to constructor
- adds checkSignature method in parseHook

Related to https://github.com/screwdriver-cd/screwdriver/pull/313

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -479,6 +481,22 @@ class GithubScm extends Scm {
...
+     * @method _getWebhookStrategy
+     * @return {Object}             Strategy object with required interface plugin, scheme, and any options needed
+     */
+    _getWebhookStrategy() {
+        return {
+            // eslint-disable-next-line global-require
+            plugin: require('hapi-github-webhooks'),
+            scheme: 'githubwebhooks',
+            options: {
+                secret: this.secret
```
Comment (minz1027) [inline index.js]: this should be this.config.secret

**Comment by minz1027** [comment]
@stjohnjohnson Not sure if secret should be optional or required. It is optional in github. Any suggestions?

File: index.js [hunk 2]
```diff
@@ -375,6 +379,29 @@ class GithubScm extends Scm {
...
+     * @param   {String}    secret      The secret used to sign the payload
+     * @param   {String}    payload     The payload of the webhook event
+     * @param   {String}    signature   The signature of the webhook event
+     * @returns {boolean}
+     */
+    _checkSignature(secret, payload, signature) {
+        const hmac = crypto.createHmac('sha1', secret);
+
+        hmac.setEncoding('hex');
+        hmac.write(JSON.stringify(payload), 'utf-8');
```
Comment (stjohnjohnson) [inline index.js]: Interesting, I wonder if this is the same as hmac.write(new Buffer(text, 'utf-8'));

File: index.js [hunk 3]
```diff
@@ -375,6 +379,29 @@ class GithubScm extends Scm {
...
+        hmac.end();
+
+        const sha = hmac.read();
+        const hash = `sha1=${sha}`;
+        const hashBuffer = new Buffer(hash);
+        const digestBuffer = new Buffer(signature);
+
+        return bufferEq(hashBuffer, digestBuffer);
```
Comment (stjohnjohnson) [inline index.js]: Honestly, I was never certain why they did this.  === of strings should be fine.  You don't need to compare buffers.

File: index.js (hunk 2 above)
Comment (minz1027) [inline index.js]: Just copy over this
https://github.com/mhazy/hapi-github-webhooks/blob/master/lib/index.js#L55-L56
I install this locally and the test it out with our actual signing secret, it works fine :)

File: index.js (hunk 3 above)
Comment (minz1027) [inline index.js]: fixed and tested

**Comment by stjohnjohnson** [comment]
@minz1027 required works for me. It's adding security.

**Pull request merged**